tests/consensus_tests/test_partial_snapshot.py

```
@@ -31,6 +32,34 @@ def test_partial_snapshot(tmp_path: pathlib.Path, bootstrap_points: int, recover
+    assert_project_root()
+
+    write_peer, read_peer = bootstrap_peers(tmp_path, 100_000)
+
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers = 3)
+    futures = [executor.submit(try_recover_partial_snapshot_from, read_peer, write_peer, wait = False) for _ in range(3)]
+    responses = [future.result() for future in concurrent.futures.as_completed(futures)]
+
+    assert any(response.status_code == 400 for response in responses), "Subsequent partial snapshot recovery requests have to be rejected during partial snapshot recovery"
+
+def test_partial_snapshot_read_lock(tmp_path: pathlib.Path):
+    assert_project_root()
```

Ruff (0.11.9)

`assert_project_root` may be undefined, or defined from star imports

(F405)